### ISSUE TITLE
Load Existing Worlds Whose Plugin No Longer Exists

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/SpongeCommon#34]
So, in the past, Bukkit functionality had maintained that a world could be loaded by an external plugin, such as [Multiverse](http://dev.bukkit.org/bukkit-plugins/multiverse-core/), the world would only be loaded if Multiverse was running on the server, otherwise, only the default world as defined by the `server.properties` file would be loaded. In SpongeAPI implementations, this isn't the case, if a plugin adds a new world, that world will be loaded every server start afterwards, in other words, that world information persists to be loaded regardless whether the plugin requesting that world is loaded by the server or not.

So, my question is simple: Do we want to have implementation where a world is only loaded if the plugin requesting that world is existing on the server? Or do we want to retain loaded worlds once they are requested, only to be removed by the request of a similar "world management" plugin?

**Comment by Zidane** [comment]
@gabizou 

We discussed this at length last night. I will say that when @bloodmc first got the world changes working in Sponge and I removed the plugin, it surprised me when they still loaded.

The beauty of this is it isn't too...difficult to support this as we can simply track if a world is set to load and then only load worlds that fit this criteria. Question is, should this be the expected behavior or leave as-is?

**Comment by gabizou** [comment]
> Question is, should this be the expected behavior or leave as-is?

That's why I created the issue so we can get some feedback from other people ;)

**Comment by Mumfrey** [comment]
Maybe actually allow the plugin to control this? Plugin creating world can say "load this even if I'm not present" or "only load if I am present". Could even be as simple as marking the world with an "owner" (plugin), which if present is required to load the world, and if left blank (by the seeding plugin) will allow the world to load regardless.

The import of this issue resolves to plugins which provide world generation features. A plugin should be able to say "loading this world without me present will cause bad chunks/weird shit" vs. a plugin saying, "I'm creating this, it's a regular world in all other respects so go ahead and load it regardless.

tl;dr If in doubt, make it configurable.

**Comment by boformer** [comment]
If you want provide multi-world support, do it right: Add a worlds config and maybe some commands to load/create/unload worlds.

Or follow the vanilla behavior: Without plugins, only load one world.

**Comment by ST-DDT** [comment]
Yes. World should only be loaded if the server is requested to do so. Either by config oder method call.

**Comment by JBYoshi** [comment]
I agree with Mumfrey - allow the `WorldBuilder` to set this.

**Comment by TheRaspPie** [comment]
@JBYoshi
- you can't easily see which world requires which plugin, which is bad for transparency
- you shouldn't mix the concerns of when to load a world with the actual content of it
- personally, I think it's ugly if you have a server writing any data other than Vanilla to a world folder

**Comment by Zidane** [comment]
Closing, admins have full control over loading a world and it is up to them if they let a world load when a vital plugin/mod is missing.
